src/wallet/sqlite.cpp

```diff
@@ -319,7 +329,8 @@ void SQLiteDatabase::Open()
     }
 
     // Do the db setup things because the table doesn't exist only when we are creating a new wallet
-    if (!table_exists) {
+    // Only do table creation and setup in read-write mode
+    if (!table_exists && !m_read_only) {
```

Comment:
In f6f1f4199b4a3c2d7ef7c1bcfa122a9d8c5eeb64 "wallet: Allow read-only database access for info and dump commands"

If the table doesn't exist and we are in read-only mode, we should throw an exception because nothing else is going to work here. We expect the "main" table to exist for any read operations.

Reply:
You mean we should throw an exception before get a return from `sqlite3_exec` ?